src/containers/blocks.jsx

```diff
@@ -284,14 +284,17 @@ class Blocks extends React.Component {
     onWorkspaceMetricsChange () {
         const target = this.props.vm.editingTarget;
         if (target && target.id) {
-            const workspaceMetrics = Object.assign({}, this.state.workspaceMetrics, {
-                [target.id]: {
+            // Dispatch updateMetrics later, since onWorkspaceMetricsChange may be (very indirectly)
```

Comment:
Does this look kosher to others?

Reply:
Got feedback from others -- sounds like this is ok, though there is some feeling that it might not be necessary.

I assume it's necessary for the reasons @towerofnix wrote in the comment -- that you don't want to dispatch a reducer action while already handling a reducer action.